assets/javascript/script.js

```diff
@@ -84,33 +77,80 @@ $(document).ready(function () {
         const errors = JSHINT.data().errors;
         result = [];
         if (errors && errors.length) {
-            $.each(errors, function (index, item) {
+            $.each(errors, function(index, item) {
                 createAndPushErrorObject(item.line, item.reason, item.id, item.evidence);
-
             });
         }
     }
 
+    function validateHtml(html) {
+        // emulate form post
+        var formData = new FormData();
+        formData.append('out', 'json');
+        formData.append('content', html);
+
+        // make ajax call
+        $.ajax({
+            url: "https://html5.validator.nu/",
+            data: formData,
+            dataType: "json",
+            type: "POST",
+            processData: false,
+            contentType: false,
+            success: function(result) {
+                const errors = result.messages;
+                result = [];
+                if (errors && errors.length) {
+                    $.each(errors, function(index, item) {
+                        createAndPushErrorObject(item.lastLine, item.message, item.type, item.extract);
+                    });
+                    renderResult();
+                }
+            }
+        });
+    }
+
+    function validateCss(inputtedCodeToBeValidated) {
+        const outputFormat = "text/plain";
+        const url = `https://cors-anywhere.herokuapp.com/http://jigsaw.w3.org/css-validator/validator?text=${encodeURIComponent(inputtedCodeToBeValidated)}&warning=0&profile=css2&output=${encodeURIComponent(outputFormat)}`;
+        // make ajax call
+        $.ajax({
+                url,
+                method: "GET"
+            })
+            .then(function(response) {
+                console.log(response);
+            });
+    }
+
     /**
      * prepares error object using input fields and push it to result array.
      * @param {line number where error is being reported} lineNum 
-     * @param {error reason} reason 
-     * @param {severity of error} severity 
-     * @param {line number content} evidence 
+     * @param {error reason or message} reason 
+     * @param {severity or type of error} severity 
+     * @param {line of code that has error} evidence 
      */
     function createAndPushErrorObject(lineNum, reason, severity, evidence) {
         const error = {};
         if (lineNum) {
-            error['lineNo'] = lineNum;
+            error.lineNo = lineNum;
+        } else if (lineNum) {
```

Comment:
We can remove the "else if" for each error object and the code will still function :)